plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1515,7 +1515,15 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         }
 
         RewriteFiles rewriteFiles = transaction.newRewrite();
-        rewriteFiles.rewriteFiles(scannedDataFiles, fullyAppliedDeleteFiles, newFiles, ImmutableSet.of());
+        for (DataFile dataFile: scannedDataFiles) {
+            rewriteFiles.deleteFile(dataFile);
+        }
+        for (DeleteFile deleteFile: fullyAppliedDeleteFiles) {
+            rewriteFiles.deleteFile(deleteFile);
+        }
+        for (DataFile dataFile: newFiles) {
+            rewriteFiles.addFile(dataFile);
+        }
```

Comment:
A space ` ` before `:` is missing. We could use `forEach` instead:
```java
        scannedDataFiles.forEach(rewriteFiles::deleteFile);
        fullyAppliedDeleteFiles.forEach(rewriteFiles::deleteFile);
        newFiles.forEach(rewriteFiles::addFile);
```

Reply:
I haven't checked Checkstyle errors 🙇‍♂️